samples/apps/tpcc/app/tpcc.cpp

```diff
@@ -0,0 +1,619 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#include "enclave/appinterface.h"
+#include "node/rpc/userfrontend.h"
+#include "tpcc_entities.h"
+#include <ctime>
+
+using namespace ccf;
+
+namespace ccfapp
+{
+namespace tpcc
+{
+
+  struct Procs {
+    static constexpr auto TPCC_NEW_ORDER = "TPCC_new_order";
+    static constexpr auto TPCC_QUERY_ORDER_HISTORY = "TPCC_query_order_history";
+    
+    static constexpr auto TPCC_LOAD_ITEMS = "TPCC_load_items";
+    static constexpr auto TPCC_LOAD_WAREHOUSE = "TPCC_load_warehouse";
+    static constexpr auto TPCC_LOAD_STOCKS = "TPCC_load_stocks";
+    static constexpr auto TPCC_LOAD_DISTRICT = "TPCC_load_district";
+    static constexpr auto TPCC_LOAD_CUSTOMER = "TPCC_load_customer";
+    static constexpr auto TPCC_LOAD_HISTORY = "TPCC_load_history";
+    static constexpr auto TPCC_LOAD_ORDER = "TPCC_load_order";
+    static constexpr auto TPCC_LOAD_ORDER_LINES = "TPCC_load_order_lines";
+    static constexpr auto TPCC_LOAD_NEW_ORDERS = "TPCC_load_new_orders";
+  };
+
+  struct TpccTables
+  {
+    Store::Map<WarehouseId, Warehouse>& warehouses;
+    Store::Map<DistrictId, District>& districts;
+    Store::Map<CustomerId, Customer>& customers;
+    Store::Map<HistoryId, History>& histories;
+    Store::Map<NewOrderId, NewOrder>& neworders;
+    Store::Map<OrderId, Order>& orders;
+    Store::Map<OrderLineId, OrderLine>& orderlines;
+    Store::Map<ItemId, Item>& items;
+    Store::Map<StockId, Stock>& stocks;
+  
+    TpccTables(Store& store) :
+      warehouses(store.create<WarehouseId, Warehouse>("warehouses")),
+      districts(store.create<DistrictId, District>("districts")),
+      customers(store.create<CustomerId, Customer>("customers")),
+      histories(store.create<HistoryId, History>("histories")),
+      neworders(store.create<NewOrderId, NewOrder>("neworders")),
+      orders(store.create<OrderId, Order>("orders")),
+      orderlines(store.create<OrderLineId, OrderLine>("orderlines")),
+      items(store.create<ItemId, Item>("items")),
+      stocks(store.create<StockId, Stock>("stocks"))
+    {}
+  };
+
+  class TpccHandlers: public UserHandlerRegistry
+  {
+  private:
+    TpccTables tables;
+
+  public:
+    TpccHandlers(Store& store) : UserHandlerRegistry(store), tables(store)
+    {}
+
+    void init_handlers(Store& store) override
+    {
+      UserHandlerRegistry::init_handlers(store);
+
+      auto queryOrderHistory = [this](Store::Tx& tx, const nlohmann::json& params) {
+        std::cout << "Processing history query..." << std::endl;
```

Comment:
you should use LOG_INFO or LOG_INFO_FMT

when using std::cout or printf this will result an ocall. The methods mentioned above will print your logs without having to perform and ocall. The logs will go to the "out" file